core/impl/src/main/java/com/blazebit/persistence/impl/function/window/AbstractWindowFunction.java

```diff
@@ -240,7 +240,7 @@ protected void render(FunctionRenderContext context, WindowFunction windowFuncti
             context.addChunk(" ");
         }
         renderOrderBy(context, windowFunction.getOrderBys());
-        if (windowFunction.getFrameMode() != null && !windowFunction.getOrderBys().isEmpty()) {
+        if (windowFunction.getFrameMode() != null && (!windowFunction.getOrderBys().isEmpty() || !windowFunction.getPartitionExpressions().isEmpty())) {
```

Comment:
The issue is a bit artificial, because one probably shouldn't use an order by without a partition by clause, but if one does, the space wasn't added resuling in an error (because it renders `ORDER BY columnnameROWS` in SQL)